cmd/cstor-volume-mgmt/volume/volume.go

```diff
@@ -104,90 +178,22 @@ func extractReplicaStatusFromJSON(str string) (*apis.CVStatus, error) {
 }
 
 // CreateIstgtConf creates istgt.conf file
-func CreateIstgtConf(cStorVolume *apis.CStorVolume) []byte {
-
-	var buffer bytes.Buffer
-	buffer.WriteString(`# Global section
-[Global]
-`)
-	buffer.WriteString("  NodeBase \"" + cStorVolume.Spec.NodeBase + "\"")
-	buffer.WriteString(`
-  PidFile "/var/run/istgt.pid"
-  AuthFile "/usr/local/etc/istgt/auth.conf"
-  LogFile "/usr/local/etc/istgt/logfile"
-  Luworkers 6
-  MediaDirectory "/mnt"
-  Timeout 60
-  NopInInterval 20
-  MaxR2T 16
-  DiscoveryAuthMethod None
-  DiscoveryAuthGroup None
-  MaxSessions 32
-  MaxConnections 4
-  FirstBurstLength 262144
-  MaxBurstLength 1048576
-  MaxRecvDataSegmentLength 262144
-  MaxOutstandingR2T 16
-  DefaultTime2Wait 2
-  DefaultTime2Retain 20
-  OperationalMode 0
-
-# UnitControl section
-[UnitControl]
-  AuthMethod None
-  AuthGroup None
-`)
-	buffer.WriteString("  Portal UC1 " + cStorVolume.Spec.TargetIP + ":3261\n")
-	buffer.WriteString("  Netmask " + cStorVolume.Spec.TargetIP + "/8\n")
-	buffer.WriteString(`
-# PortalGroup section
-[PortalGroup1]
-`)
-	buffer.WriteString("  Portal DA1 " + cStorVolume.Spec.TargetIP + ":3260\n")
-	buffer.WriteString(`
-# InitiatorGroup section
-[InitiatorGroup1]
-  InitiatorName "ALL"
-  Netmask "ALL"
-
-[InitiatorGroup2]
-  InitiatorName "None"
-  Netmask "None"
-
-# LogicalUnit section
-[LogicalUnit1]
-`)
-	buffer.WriteString("  TargetName " + cStorVolume.Name + "\n")
-	buffer.WriteString("  TargetAlias nicknamefor-" + cStorVolume.Name)
-	buffer.WriteString(`
-  Mapping PortalGroup1 InitiatorGroup1
-  AuthMethod None
-  AuthGroup None
-  UseDigest Auto
-  ReadOnly No
-`)
-	buffer.WriteString("  ReplicationFactor " + strconv.Itoa(cStorVolume.Spec.ReplicationFactor) + "\n")
-	buffer.WriteString("  ConsistencyFactor " + strconv.Itoa(cStorVolume.Spec.ConsistencyFactor))
-	buffer.WriteString(`
-  UnitType Disk
-  UnitOnline Yes
-  BlockLength 512
-  QueueDepth 32
-  Luworkers 6
-`)
-	buffer.WriteString("  UnitInquiry \"OpenEBS\" \"iscsi\" \"0\" \"" + string(cStorVolume.UID) + "\"")
-	buffer.WriteString(`
-  PhysRecordLength 4096
-`)
-	buffer.WriteString("  LUN0 Storage " + cStorVolume.Spec.Capacity + " 32k")
-	buffer.WriteString(`
-  LUN0 Option Unmap Disable
-  LUN0 Option WZero Disable
-  LUN0 Option ATS Disable
-  LUN0 Option XCOPY Disable
-`)
-
-	return buffer.Bytes()
+func CreateIstgtConf(cStorVolume *apis.CStorVolume) ([]byte, error) {
```

Comment:
@shubham14bajpai can you review this template code?

Reply:
sure